[08:12] *** sena_kun joined
[08:46] *** sena_kun left
[14:03] <lizmat> hmmmm in my work on ParaSeq, I'm seeing occasional test failures that can only explain by nqp::eqaddr() returning something other than 0 or 1

[14:05] <lizmat> nvm, it's probably something else

[17:51] *** sena_kun joined
[20:48] <Geth> ¦ MoarVM/main: 51ca012d20 | (Patrick Böker)++ | src/core/frame.c

[20:48] <Geth> ¦ MoarVM/main: Reapply "Don't lose return value with LEAVE phasers"

[20:48] <Geth> ¦ MoarVM/main: 

[20:48] <Geth> ¦ MoarVM/main: This reverts commit d67da80f392eff0c262491c05520bbc3cdc93eb9.

[20:48] <Geth> ¦ MoarVM/main: 

[20:48] <Geth> ¦ MoarVM/main: 2024.05 has happened. So give this another try.

[20:48] <Geth> ¦ MoarVM/main: review: https://github.com/MoarVM/MoarVM/commit/51ca012d20

[20:49] <patrickb> Any reason not to also merge Idea: Let's distinguish `CONTROL` exceptions from `CATCH` exceptions.

[20:49] <patrickb> That's an already well established distinction.

[20:49] <patrickb> Given the word "conflict" means: An exception thrown within a phaser

[20:49] <patrickb> during an unwind (a `LEAVE`) wants to unwind out of that phasers

[20:49] <patrickb> frame.

[20:49] <patrickb> Then I propose the following rules:

[20:49] <patrickb> - A `CATCH` exception is in progress, a `CONTROL` exception conflicts.

[20:49] <patrickb> -> We ignore the `CONTROL` exception and continue with the `CATCH`

[20:50] <patrickb> exceptions unwind.

[20:50] <patrickb> - A `CONTROL` exception is in progress, a `CONTROL` exception

[20:50] <patrickb> conflicts. (That's what has been discussed at length in this issue.)

[20:50] <patrickb> -> We compare the target frames of the two exception handlers. The one

[20:50] <patrickb> unwinding more of the stack wins. If it's a tie, the later one (the

[20:50] <patrickb> one originating from within the phaser) wins. (This behavior is

[20:50] <patrickb> implemented in https://github.com/MoarVM/MoarVM/pull/1812.)

[20:50] <patrickb> - A `CONTROL` or `CATCH` exception is in progress, a `CATCH` exception

[20:50] <patrickb> conflicts. -> We add the exception to a list (located in the frame the

[20:50] <patrickb> current unwind targets). After the unwind finishes and before we run

[20:50] <patrickb> the handler, we throw all the collected exceptions. That's what

[20:50] <patrickb> [S04](https://github.com/Raku/old-design-docs/blob/a4c36c683dafdad0bb55996e78b66a1f48fc703c/S04-control.pod#L1585-L1588)

[20:50] <patrickb> arg! Sorry, misspaste.

[20:50] <patrickb> Any reason not to also merge https://github.com/MoarVM/MoarVM/pull/1785 ?

[20:50] <patrickb> This is what it should have been. /o\

[22:33] *** sena_kun left
